src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder option is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Stops recording a given sensor type.
+        /// </summary>
+        /// <param name="type">
+        /// A sensor type to be recorded.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void Stop(int type)
+        {
+            int error = Interop.SensorRecoder.Stop(type);
+            if (error != (int)SensorError.None)
+            {
+                Log.Error(Globals.LogTag, "Error in stopping recorder");
+                throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.Stop Failed");
+            }
+
+        }
+
+        /// <summary>
+        /// Destroys a recorder option handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void DestroyOption()
+        {
+            if (CheckOption())
+            {
+                int error = Interop.SensorRecoder.DestroyOption(OptionHandle);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error deleting the sensor recorder option");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.DestroyOption Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder option is not created");
+            }
+        }
+
+        /// <summary>
+        /// Destroys a recorder query handle.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        private void DestroyQuery()
+        {
+            if (CheckQuery())
+            {
+                int error = Interop.SensorRecoder.DestroyQuery(QueryHandle);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error deleting the sensor recorder query");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.DestroyQuery Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        /// Overloaded Dispose API for destroying the SensorProvider Handle.
+        /// </summary>
+        /// <since_tizen> 3 </since_tizen>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Dispose API for destroying the SensorProvider handle.
+        /// </summary>
+        /// <since_tizen> 3 </since_tizen>
+        /// <param name="disposing">The boolean value for destoying SensorProvider handle.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
```

Comment:
You should only release managed resource when disposing was true. 
You should alway release unmanaged resource regardless of disposing value